synth/src/cli/mongo.rs

```diff
@@ -232,6 +233,54 @@ impl MongoExportStrategy {
     }
 }
 
+fn value_to_bson(value: Value) -> Bson {
+    match value {
+        Value::Null(_) => Bson::Null,
+        Value::Bool(b) => Bson::Boolean(b),
+        Value::Number(n) => number_to_bson(n),
+        Value::String(s) => Bson::String(s),
+        Value::DateTime(dt) => date_time_to_bson(dt),
+        Value::Object(obj) => object_to_bson(obj),
+        Value::Array(arr) => array_to_bson(arr)
+    }
+}
+
+fn array_to_bson(array: Vec<Value>) -> Bson {
+    Bson::Array(array.into_iter().map(value_to_bson).collect())
+}
+
+fn object_to_bson(obj: BTreeMap<String, Value>) -> Bson {
+    let obj = obj.into_iter().map(|(name, value)| (name, value_to_bson(value))).collect();
+    Bson::Document(obj)
+}
+
+
+fn date_time_to_bson(datetime: ChronoValue) -> Bson {
+    match datetime {
```

Comment:
Actually by that logic the `to_json` for `Value::DateTime` should `Err`'out (because it is not representable in the `JSON::Value` format). However, we do want to format it using the user specified format. So in a sense these conversion mappings between `Graph::Value` and data sinks should also support user-specified formatting.

Reply:
I don't think `Err`ing here will give us much benefit, but I agree that formatting to a string is not a great solution here; changing all variants to a `Bson::DateTime` (using MIN_DATE for plain `Time`s for now).

We should probably document this, though.